### PR TITLE
Nightmare re-inject scripts

### PR DESCRIPTION
**Pull request by APshenkin**
What is in this PR:
* Fix https://github.com/Codeception/CodeceptJS/issues/401
* Add more detailed error messages for Wait methods for Nightmare
* Fix adding screenshots to Mochawesome in Nightmare
* Fix unique screenshots names in Nightmare
* Fix Mochawesome context for failed screenshots. At this moment you have to put html report on the same folder, where you store screenshots
* Fix CodeceptJS work with hooks in helpers. Now we will finish codeceptJS correctly, if some errors appears in helpers hooks
* If we receive selenium grid error when use `restart:false`, then we will create a new session for next test
* Now CodeceptJS will create screenshots for failed hooks from Feature file
* Fix https://github.com/Codeception/CodeceptJS/issues/639
* Fix https://github.com/Codeception/CodeceptJS/issues/644

### CODE REVIEW

File: lib/listener/helpers.js [hunk 1]
```diff
@@ -29,11 +29,27 @@ module.exports = function () {
...
-    runAsyncHelpersHook('_beforeSuite', suite, true);
+    try {
+      runAsyncHelpersHook('_beforeSuite', suite, true);
+    } catch (err) {
+      recorder.throw(err);
+    }
+    recorder.catch((err) => {
+      event.emit(event.test.failed, suite, err); // emit
+      throw err;
+    });
```
Comment (DavertMik) [inline lib/listener/helpers.js]: code duplication. Should be wrapped into another function and reused for suite.before, suite.after, test.before

**Comment by DavertMik** [comment]
Ok, one minor stuff and I'm ready to merge

File: lib/scenario.js [hunk 2]
```diff
@@ -80,45 +80,83 @@ module.exports.test = (test) => {
...
+    recorder.startUnlessRunning();
+    event.emit(event.suite.after, suite);
+  } catch (err) {
+    recorder.throw(err);
+  }
+  recorder.catch((err) => {
+    event.emit(event.test.failed, suite, err);
+    throw err;
+  });
+  return recorder.promise();
```
Comment (DavertMik) [inline lib/scenario.js]: still the same: please make one function and reuse it for suiteTeardown, suiteSetup and setup

Comment (APshenkin) [inline lib/scenario.js]: fixed

**Comment by DavertMik** [comment]
Restarted job on Travis CI. Let's wait for it

**Comment by mracz** [comment]
@DavertMik Do you plan to release a new version with these changes? I would really need the a fix for https://github.com/Codeception/CodeceptJS/pull/635/commits/9808b1b1a22137b469db664e7489a8ec342f4bad.
Thanks for your answer in advance!

**Comment by BrandonWade** [comment]
@DavertMik A new release would be really appreciated! I too need the fix for 9808b1b.

**Comment by APshenkin** [comment]
@mracz @BrandonWade New release is already pushed. Check http://codecept.io/changelog/

**Comment by BrandonWade** [comment]
Much appreciated. Unfortunately I'm still getting the error when using waitForElement described here:
https://github.com/Codeception/CodeceptJS/issues/401

**Comment by APshenkin** [comment]
@BrandonWade 
Please create a new issue for it with reproducible steps.

**Pull request merged**